api/main.py

```diff
@@ -368,3 +371,120 @@ class parallelItem(BaseModel):
     except KeyError as e:
         print("KeyError: ", e)
         raise HTTPException(status_code=400)
+
+
+@app.get("/visual/{searchterm}")
+async def get_graph_for_file(
+    searchterm: str,
+    language: str,
+    selected: List[str] = Query([]),
+):
+    try:
+        db = get_db()
+        total_parallel_count = []
+        searchtype = 'category'
+        if re.search("^[A-Z][a-z]+$", searchterm):
+            searchtype = "collection"
+
+        # get a sorted list of categories to get the results in the right order
+        query_full_category_list = db.AQLQuery(
+            query=query_sorted_category_list,
+            bindVars={
+                "language": language,
+                "selected": selected,
+            },
+        )
+        category_list = {}
+        for category in query_full_category_list.result:
+            category_list.update(category)
```

Comment:
hmm, what is for for? Can't we just assign the result to the object directly without iterating?

Reply:
Maybe you know a better solution for this. The output is in the form of a list of dictionaries with each dictionary just one item in it in the right order. So I want to get that to one dictionary that keeps the same order. Maybe best to use OrderedDict instead. What do you think?

Reply:
I see, now it makes sense. Dicts are only guaranteed to be insertion-ordered since Python 3.7, but technically we can use this.

Can we just rename `category_list` since it's not a list?